From c576a271b82678f50fb1e8067ae975015a8b9059 Mon Sep 17 00:00:00 2001
Message-Id: <c576a271b82678f50fb1e8067ae975015a8b9059.1426574757.git.chang-joon.lee@intel.com>
In-Reply-To: <eba22108096acc90a0516b200c616db90ae0cb5f.1426574757.git.chang-joon.lee@intel.com>
References: <eba22108096acc90a0516b200c616db90ae0cb5f.1426574757.git.chang-joon.lee@intel.com>
From: Hong Liu <hong.liu@intel.com>
Date: Sat, 28 Feb 2015 16:12:03 +0800
Subject: [PATCH 2/8] REVERTME [VPG]: drm/i915: fix no display in charging OS

If we boot to COS/ROS, we need to do a first modeset to keep driver
having proper state updated.

And later in COS/ROS, we just need to call encoder enable/disable to
turn on/off display.

REVERTME: This patch can be reverted once we figure out a way to correctly
setup all the driver state after driver loading.

Issue: IMINAN-21672
Change-Id: I64535870d839949620ff0d54ec96fd2d5b0a8c71
Signed-off-by: Hong Liu <hong.liu@intel.com>
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |   11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

Index: linux/drivers/gpu/drm/i915/intel_display.c
===================================================================
--- linux.orig/drivers/gpu/drm/i915/intel_display.c	2015-03-20 10:55:24.403745736 -0700
+++ linux/drivers/gpu/drm/i915/intel_display.c	2015-03-20 10:58:09.137728286 -0700
@@ -5982,8 +5982,15 @@
 		intel_modeset_setup_hw_state(dev, true);
 
 	/* Only need to change hw state when actually enabled */
-	if (connector->encoder)
-		intel_encoder_dpms(to_intel_encoder(connector->encoder), mode);
+	if (connector->encoder) {
+		if (mode == DRM_MODE_DPMS_ON && dev_priv->is_first_modeset) {
+			intel_modeset_setup_hw_state(dev, true);
+			dev_priv->is_first_modeset = false;
+		} else {
+			intel_encoder_dpms(to_intel_encoder(connector->encoder),
+					   mode);
+		}
+	}
 
 	if (mode == DRM_MODE_DPMS_ON)
 		intel_modeset_check_state(connector->dev);
